document-generator-core/src/test/java/uk/gov/companieshouse/document/generator/core/service/DocumentGeneratorServiceTest.java

```diff
@@ -0,0 +1,152 @@
+        assertNotNull(response.getDescription());
+        assertNotNull(response.getDescriptionIdentifier());
+        assertNotNull(response.getDescriptionValues());
+        assertNotNull(response.getLocation());
+        assertNotNull(response.getSize());
+    }
+
+    @Test
+    @DisplayName("Tests when null returned from documentInfoService")
+    public void testsWhenErrorThrownFromDocumentInfoService() {
+
+        when(mockDocumentInfoService.getDocumentInfo(any(DocumentInfoRequest.class))).thenReturn(null);
+
+        DocumentResponse response = documentGeneratorService.generate(setValidRequest());
+
+        assertNull(response);
+    }
+
+    @Test
+    @DisplayName("Tests when an error thrown from requestHandler")
+    public void testsWhenErrorThrownFromRequestHandler() throws IOException {
+
+        when(mockDocumentInfoService.getDocumentInfo(any(DocumentInfoRequest.class))).thenReturn(setSuccessfulDocumentInfo());
+
+        when(mockRequestHandler.sendDataToDocumentRenderService(any(String.class), any(RenderDocumentRequest.class))).thenThrow(IOException.class);
+
+        DocumentResponse response = documentGeneratorService.generate(setValidRequest());
+
+        assertNotNull(response);
+        assertNotNull(response.getDescription());
+        assertNotNull(response.getDescriptionIdentifier());
+        assertNotNull(response.getDescriptionValues());
+        assertNull(response.getLocation());
+        assertNull(response.getSize());
+    }
+
+    /**
+     * Set data for valid DocumentRequest
+     *
+     * @return
+     */
+    private DocumentRequest setValidRequest() {
+
+        DocumentRequest request = new DocumentRequest();
+        request.setContentType("contentType");
+        request.setDocumentType("documentType");
+        request.setId("id");
+        request.setResource("resource");
+        request.setResourceId("resourceId");
+        request.setUserId("userId");
+
+        return request;
+    }
+
+    /**
+     * Set Data for successful RenderResponse
+     *
+     * @return
+     */
+    private RenderDocumentResponse setSuccessfulRenderResponse() {
+
+        RenderDocumentResponse renderDocumentResponse = new RenderDocumentResponse();
+        renderDocumentResponse.setDocumentSize("size");
+        renderDocumentResponse.setLocation("location.file");
+
+        return renderDocumentResponse;
+    }
+
+    /**
+     * Set data from successful DocumentInfoService call
+     *
+     * @return
+     */
+    private DocumentInfoResponse setSuccessfulDocumentInfo() {
+
+        DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
+        documentInfoResponse.setLocation("location");
+        documentInfoResponse.setTemplateName("templateName");
+        documentInfoResponse.setData("data");
+        documentInfoResponse.setAssetId("assetId");
+        documentInfoResponse.setDescription("description");
+        documentInfoResponse.setDescriptionIdentifier("descriptionIdentifier");
```

Comment:
Can you make these into constants and verify with an assertEquals() instead of assertNull() in the tests please?